src/crypto/Signature.ts

```diff
@@ -1,13 +1,14 @@
-import { xmtpEnvelope as proto } from '@xmtp/proto'
+import { signature } from '@xmtp/proto'
 import Long from 'long'
 import * as secp from '@noble/secp256k1'
 import PublicKey from './PublicKey'
 
 // Signature represents an ECDSA signature with recovery bit.
-export default class Signature implements proto.Signature {
-  ecdsaCompact: proto.Signature_ECDSACompact | undefined // eslint-disable-line camelcase
+export default class Signature implements signature.Signature {
+  ecdsaCompact: signature.Signature_ECDSACompact | undefined // eslint-disable-line camelcase
+  walletEcdsaCompact: signature.Signature_WalletECDSACompact | undefined // eslint-disable-line camelcase
 
-  constructor(obj: proto.Signature) {
+  constructor(obj: Partial<signature.Signature>) {
```

Comment:
Why the Partial?

Reply:
Great question. Without the `Partial` it expects both the `ecdsaCompact` and the `walletEcdsaCompact`. The alternative is to make any caller always provide both with one undefined.

Reply:
Ya, it's one of Typescript's weird quirks. https://www.becomebetterprogrammer.com/typescript-question-mark/